### PR TITLE
migrate scripts to yq4

### PR DESCRIPTION
**Pull request by xrstf**
**What does this PR do / Why do we need it**:
It's time to migrate to yq4. See https://mikefarah.gitbook.io/yq/upgrading-from-v3 for more information.

I chose to make it explicit what version we're using for now, in order to slowly get rid of yq3 over the next few PRs. It is a bit annoying that devs would need a  `yq4` binary on their machines, but I suspect many have v3 and v4 around, like I do.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Review by embik — APPROVED** [review]

File: hack/ci/fetch-chart-dependencies.sh [hunk 1]
```diff
@@ -28,11 +28,11 @@ charts=$(find charts/ -name Chart.yaml | sort)
 
 [ -n "$charts" ] && while read -r chartYAML; do
   dirname="$(dirname $(echo "$chartYAML"))"
-  chartname=$(yq read "$chartYAML" name)
+  chartname=$(yq4 '.name' "$chartYAML")
   echodate "Fetching dependencies for ${chartname}..."
 
   i=0
-  for url in $(yq r "$chartYAML" dependencies --tojson | jq -r .[].repository); do
+  for url in $(yq4 '.dependencies.[].repository' Chart.yaml); do
```
Comment (xmudrii) [inline hack/ci/fetch-chart-dependencies.sh]: Shouldn't this be `$chartYAML`?

**Review by xmudrii — COMMENTED** [review]
Just a tiny nit.

File: hack/ci/fetch-chart-dependencies.sh (hunk 1 above)
Comment (xrstf) [reply hack/ci/fetch-chart-dependencies.sh]: It should.

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Pull request merged**